canvas/src/layer/layer_2d.rs

```diff
@@ -3,32 +3,161 @@ use crate::{
+  #[inline]
+  pub fn begin_path(&mut self, x: f32, y: f32) -> &mut Self {
+    self.path = Some(Builder::new());
+    self.path.as_mut().map(|b| b.begin(point(x, y)));
+    self
+  }
+
+  /// Causes the point of the pen to move back to the start of the current
+  /// sub-path. It tries to draw a straight line from the current point to the
+  /// start. If the shape has already been closed or has only one point, this
+  #[inline]
+  pub fn close_path(&mut self) -> &mut Self {
+    self.path.as_mut().map(|b| b.close());
+    self
+  }
+
+  /// Connects the last point in the current sub-path to the specified (x, y)
+  /// coordinates with a straight line.
+  #[inline]
+  pub fn line_to(&mut self, x: f32, y: f32) -> &mut Self {
+    self.path.as_mut().map(|b| b.line_to(point(x, y)));
+    self
+  }
+
+  /// Adds a cubic Bezier curve to the current path.
+  #[inline]
+  pub fn bezier_curve_to(
+    &mut self,
+    cp1x: f32,
+    cp1y: f32,
+    cp2x: f32,
+    cp2y: f32,
+    x: f32,
+    y: f32,
+  ) -> &mut Self {
+    self
+      .path
+      .as_mut()
+      .map(|b| b.cubic_bezier_to(point(cp1x, cp1y), point(cp2x, cp2y), point(x, y)));
+    self
+  }
+
+  /// Adds a quadratic Bézier curve to the current path.
+  #[inline]
+  pub fn quadratic_curve_to(&mut self, cpx: f32, cpy: f32, x: f32, y: f32) -> &mut Self {
+    self
+      .path
+      .as_mut()
+      .map(|b| b.quadratic_bezier_to(point(cpx, cpy), point(x, y)));
+    self
+  }
+
+  /// Adds a circular arc to the current path.
+  pub fn arc(
+    &mut self,
+    x: f32,
+    y: f32,
+    radius: f32,
+    start_angle: Angle,
+    end_angle: Angle,
+  ) -> &mut Self {
+    let sweep_angle = end_angle - start_angle;
+    let arc = Arc {
+      start_angle,
+      sweep_angle,
+      radii: Vector::new(radius, radius),
+      center: point(x, y),
+      x_rotation: Angle::zero(),
+    };
+    let arc_start = arc.from();
+    self.begin_path(arc_start.x, arc_start.y);
+    arc.for_each_quadratic_bezier(&mut |curve| {
+      self
+        .path
+        .as_mut()
+        .map(|b| b.quadratic_bezier_to(curve.ctrl, curve.to));
+    });
+    self.close_path();
+    self
+  }
+
+  /// Adds an arc to the current path with the given control points and radius,
+  /// connected to the previous point by a straight line.
+  pub fn arc_to(&mut self, x1: f32, y1: f32, x2: f32, y2: f32, radius: f32) -> &mut Self {
+    unimplemented!();
+  }
+
+  /// Adds an elliptical arc to the current path.
+  pub fn ellipse(
+    &mut self,
+    x: f32,
+    y: f32,
+    radiusX: f32,
+    radiusY: f32,
+    rotation: f32,
+    start_angle: f32,
+    end_angle: f32,
+    winding: Winding,
+  ) -> &mut Self {
+    unimplemented!();
+  }
+
+  /// Creates a path for a rectangle at position (x, y) with a size that is
+  /// determined by width and height.
+  pub fn rect(&mut self, x: f32, y: f32, width: f32, height: f32) -> &mut Self {
+    let tr_pt = point(x + width, y);
```

Comment:
`PathBuilder` has some basic shapes method like `add_rectangle`, does it work for you ?